test/Microsoft.DotNet.PackageInstall.Tests/ToolPackageDownloaderTests.cs

```diff
@@ -54,7 +54,8 @@ public void GivenNugetConfigInstallSucceeds(bool testMockBehaviorIsInSync)
                 verbosity: TestVerbosity,
                 versionRange: VersionRange.Parse(TestPackageVersion),
                 targetFramework: _testTargetframework,
-                isGlobalTool: true);
+                isGlobalTool: true,
+                verifySignatures: false);
```

Comment:
Do we need all these test changes to specify verifySignatures = false?  It seems to me like in the tests there won't be a `RepositorySignatureResource` or it won't specify `AllRepositorySigned`, so the downloader won't verify the signatures anyway and we shouldn't need all these test changes.

Reply:
Short answer: yes.

You can see the full list of changes needed for the test in https://github.com/dotnet/sdk/pull/39268/commits/131cff5b873a66e4cf83c5048536c13e1781250e

You are generally correct that we won't verify signatures anyway in tests, but there are two tests that were different: the tests for unlisted packages. Since local feeds can't provide unlisted packages, JL03-Yue used the package the customer provided in writing tests to ensure that dotnet tool restore can find unlisted packages if necessary, and it turns out that package does claim to be signed, but the signature isn't valid, and it fails the test.

I have to admit that it took a lot of changes to thread through verifySignatures: false to a point where it could actually affect the test, and even after I got that part to work, the tests still needed more changes to find the right rid graph json, etc. If we don't want all those changes, the only alternative proposals I'm aware of are to either find another unlisted package that's signed correctly (but I/standup yesterday didn't know how to do that) or disable the test. marcpopMSFT sounded like he was most in favor of disabling the tests and avoiding all the other changes. If that's your preference as well, I can revert those last two commits and just disable them.